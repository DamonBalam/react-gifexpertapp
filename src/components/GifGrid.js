import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
    
    const { data:images, loading } = useFetchGifs(category);

    return (
        <Fragment>
            <h3 className='animate__animated animate__fadeInRight'>{category}</h3>
            {loading && <p>Loading</p>}
            <div className='card-grid'>
                {images.map((img) => (
                    <GifGridItem key={img.id} {...img} />
                ))}
            </div>
        </Fragment>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
