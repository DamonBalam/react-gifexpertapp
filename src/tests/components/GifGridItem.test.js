import { shallow } from 'enzyme';
// import '@testing-library/jest-dom';

import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas de GifGridItem.js', () => {
    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe mostrar <GifGridItem> Correctamente', () => {
        // console.log(wrapper)
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
    });

    test('Debe tener la imagen al url y alt de los props', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Debe tener animate__fadeInRight', () => {
        const div = wrapper.find('div');

        const className = div.prop('className');

        expect( className.includes('animate__fadeInRight') ).toBe(true);
    });
});
