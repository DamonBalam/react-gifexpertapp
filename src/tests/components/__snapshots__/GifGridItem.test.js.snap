// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGridItem.js Debe mostrar <GifGridItem> Correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInRight"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
